sympy/simplify/tests/test_cse.py

```diff
@@ -130,17 +130,17 @@ def test_multiple_expressions():
     assert reduced == [x0*z, x0*w]
     l = [w*x*y + z, w*y]
     substs, reduced = cse(l)
-    rsubsts, _ = cse(reversed(l))
+    rsubsts, _ = cse(list(reversed(l)))
```

Comment:
If cse no longer works with any iterable this is a regression. You can use the `iterable` function in `sympy.core.compatibility` that you can use to check if the input is an iterable and convert it to a list first in that case.

Reply:
Actually this is a bit more complicated. You can not use iterable as e.g. Matrix is iterable and `list(Matrix([[1]])) == [1]`. I'll be looking a bit more into how to properly fix this case.

Reply:
How about `iterable(arg, exclude=(string_types, dict, NotIterable, Basic))` (the same as the default version except adding `Basic`)?

Reply:
Thanks, that worked as long as I also add MatrixBase as well.